File: hoyomi_bridge/lib/core_services/shared/o_image.dart
import 'package:freezed_annotation/freezed_annotation.dart';
import 'package:hoyomi_bridge/export.dart';
import 'package:eval_annotation/eval_annotation.dart';

part 'o_image.freezed.dart';
part 'o_image.g.dart';

@freezed
@Bind()
sealed class OImage with _$OImage {
  const factory OImage({required String src, Headers? headers, String? extra}) =
      _OImage;

  factory OImage.fromJson(Map<String, dynamic> json) => _$OImageFromJson(json);
  factory OImage.from(String src) => OImage(src: src);

  factory OImage.createFakeData() {
    return OImage(src: fake);
  }

  static final fake = "fake:";
  static final blank = 'assets/images/blank.png';
}

File: lib/core_services/interfaces/o_image.ts
import type { Headers } from "./headers"

export interface OImage {
  src: string
  headers?: Headers
  extra?: string
}

export function createOImage(
  src: string,
  headers?: Headers,
  extra?: string
): OImage {
  return { src, headers, extra }
}

export const OImageFake = "fake:"
export const OImageBlank = "assets/images/blank.png"
